bread/layout/components/form.py

```diff
@@ -478,6 +464,30 @@ def _mapwidget(
             _class="bx--form-item",
         )
 
+    WIDGET_MAPPING = {
+        forms.TextInput: TextInput,
+        # Attention: NumberInput is not the widget that is used for phone numbers. See below for handling of phone numbers
+        forms.NumberInput: TextInput,
+        forms.EmailInput: EmailInput,
+        forms.URLInput: UrlInput,
+        forms.PasswordInput: PasswordInput,
+        forms.HiddenInput: HiddenInput,
+        forms.DateInput: DatePicker,
+        forms.DateTimeInput: TextInput,  # TODO
+        forms.TimeInput: TextInput,  # TODO HIGH
+        forms.Textarea: TextArea,
+        forms.CheckboxInput: Checkbox,
+        forms.NullBooleanSelect: Select,
+        forms.SelectMultiple: MultiSelect,  # TODO HIGH
+        forms.RadioSelect: TextInput,  # TODO HIGH
+        forms.FileInput: FileUploader,
+        forms.ClearableFileInput: FileUploader,  # TODO HIGH
+        forms.MultipleHiddenInput: TextInput,  # TODO
+        forms.SplitDateTimeWidget: TextInput,  # TODO
+        forms.SplitHiddenDateTimeWidget: TextInput,  # TODO
+        forms.SelectDateWidget: TextInput,  # TODO
+    }
+
```

Comment:
Every PR improves _mapwidget a little bit more, until, one day, it is gooone :D

Reply:
😄